pennylane/noise/conditionals.py

```diff
@@ -0,0 +1,458 @@
+
+    >>> cond_func = qml.noise.op_in(["RX", "RY"])
+    >>> cond_func(qml.RX(1.23, wires=[0]))
+    True
+    >>> cond_func(qml.RZ(1.23, wires=[3]))
+    False
+    >>> cond_func([qml.RX(1.23, wires=[1]), qml.RY(4.56, wires=[2])])
+    True
+
+    Additionally, an instance of :class:`Operation <pennylane.operation.Operation>`
+    can also be provided:
+
+    >>> cond_func = qml.noise.op_in([qml.RX(1.0, "dino"), qml.RY(2.0, "rhino")])
+    >>> cond_func(qml.RX(1.23, wires=["eve"]))
+    True
+    >>> cond_func(qml.RY(1.23, wires=["dino"]))
+    True
+    >>> cond_func([qml.RX(1.23, wires=[1]), qml.RZ(4.56, wires=[2])])
+    False
+    """
+    ops = [ops] if not isinstance(ops, (list, tuple, set)) else ops
+    return OpIn(ops)
+
+
+def op_eq(ops):
+    """Builds a conditional as a boolean function for evaluating
+    if a given operation is equal to the specified operation.
+
+    Args:
+        ops (str, class, Operation): string representation, an instance or class of the operation.
+
+    Returns:
+        :class:`OpEq <pennylane.noise.OpEq>`: a boolean function that evaluates to ``True``, if a
+        given operation is equal to the specified set of operation(s) based on a comparison of
+        the type, irrespective of their wires.
+
+    **Example**
+
+    One may use ``op_eq`` with a string representation of the name of the operation:
+
+    >>> cond_func = qml.noise.op_eq("RX")
+    >>> cond_func(qml.RX(1.23, wires=[0]))
+    True
+    >>> cond_func(qml.RZ(1.23, wires=[3]))
+    False
+    >>> cond_func("CNOT")
+    False
+
+    Additionally, an instance of :class:`Operation <pennylane.operation.Operation>`
+    can also be provided:
+
+    >>> cond_func = qml.noise.op_eq(qml.RX(1.0, "dino"))
+    >>> cond_func(qml.RX(1.23, wires=["eve"]))
+    True
+    >>> cond_func(qml.RY(1.23, wires=["dino"]))
+    False
+    """
+    return OpEq(ops)
+
+
+def partial_wires(operation, *args, **kwargs):
+    """Builds a partial function based on the given operation with
+    all argument frozen except ``wires``.
+
+    Args:
+        operation (Operation, class): instance of the operation or the class
+            corresponding to operation.
+        args: Positional arguments provided in the case where the keyword argument
+            ``operation`` is a class for building the partially evaluated instance.
+        kwargs: Keyword arguments for the building the partially evaluated instance.
+            These will override any arguments present in the operation instance or ``args``.
+
+    Returns:
+        callable: a wrapper function that accepts a sequence of wires as an argument or
+        any object with ``wires`` property.
+
+    Raises:
+        ValueError: if ``args`` are provided when the given ``operation`` is an instance.
+
+    **Example**
+
+    One may give an instance of :class:`Operation <pennylane.operation.Operation>`
+    for the ``operation`` argument:
+
+    >>> func = qml.noise.partial_wires(qml.RX(1.2, [12]))
```

Comment:
Probably not super important, but it does not work with some templates:

`func = qml.noise.partial_wires(qml.Qubitization(qml.X(0), control = [1]))`

It would be the same with any template that has not "wires" directly

Reply:
Yeah, it will work with operations (and templates) that take in "wires" directly. It will be too much effort to generalize it imo.